### PR TITLE
Using FFprobe to collect media data

### PR DESCRIPTION
**Pull request by hunterhogan**
### Background issue
When running Auto-Editor on my videos, I often received the error 
>Auto-Editor got 'None' when probing fps

I didn't understand why, but while troubleshooting the issue, I noticed that ffwrapper.py uses FFmpeg to retrieve data about the file. 

### Testing FFprobe
The folks at FF* say the output of FFmpeg is not machine-readable and we should use FFprobe instead, so I tested using FFprobe to gather the file data. With FFprobe, the error immediately stopped happening and it has not returned.

### I'm a noob with Git and repositories
I have never used a code repository before, except to download packages. This entire process is new to me, and this is literally my first pull request. I apologize for mistakes I am not yet aware of and for my awkwardness.

### Generalizing the code
The hacked, ugly code I wrote so that my files would process worked perfectly on dozens of my files. So, I went back and tried to recreate the functionality of the original code. I have never used subtitle _streams_ in FFmpeg, so I am skeptical that I implemented that part correctly. 

### The core idea
(Perhaps this section should be first.)

Auto-Editor wants to collect specific data about the media file. FFprobe will supply data about the media file in multiple formats. I don't know enough to speculate which format is best. I selected JSON as FFprobe's output format. Second, I only ask FFprobe for the data Auto-Editor wants. Next, I use Python's `json.loads()` to convert the data into dictionaries that seem convenient to use. Finally, I use the dictionaries to load the values into each dataclass.

### CODE REVIEW

**Comment by WyattBlue** [comment]
Two high level points:

* Yes, using FFprobe is alright but keep the ffmpeg regex code, and add some option to switch the old way. 
* Don't add the FFprobe binaries yourself. I can't verify if there's malware or not from just a blob. I can add those myself. 

More tactically, you should use f-strings instead of + for str formatting. It's faster, cleaner, and static type checkers understand them better.

```python3
# This
log.warning(
    "The command, "
    + command_with_options
    + ", included an error from stderr: "
    + err
)
# can be rewritten as this:
log.warning(
    f"The command, {command_with_options}, included an error from stderr: {err}"
)

# Also, I'd change the message to this to shorten it down a little:
log.warning(f"Command: {command_with_options}\nIncluded in stderr: {err}")
```

I don't like all the `# initialize` comments. I would just remove them.

**Most Importantly** I'm get this crash when I run:

```
python3 -m auto_editor test cli
```

```python3
 File "/Users/wyattblue/projects/auto-editor/auto_editor/validate_input.py", line 64, in valid_input
    new_inputs.append(FileInfo(my_input, ffmpeg, log))
  File "/Users/wyattblue/projects/auto-editor/auto_editor/ffwrapper.py", line 343, in __init__
    _fps = Fraction(_fps)
  File "/usr/local/Cellar/python@3.10/3.10.4/Frameworks/Python.framework/Versions/3.10/lib/python3.10/fractions.py", line 156, in __new__
    raise ZeroDivisionError('Fraction(%s, 0)' % numerator)
ZeroDivisionError: Fraction(0, 0)
```

Running:
```
python3 -m auto_editor test all
```

also crashes because it doesn't know how to handle non-media JSON files. (The current method sets some dummy values)

You need have all the tests pass if this PR is to be merged.

**In Short**
Need revision. I need to figure how to enable auto-test for Pull Requests.

**Comment by WyattBlue** [comment]
I don't like making FFprobe a separate class, using FileInfo as a common interface makes much more since. Also your PR is not working on Linux. Probably because you're running "FFprobe" instead of "ffprobe" in subprocess.

I decided to implement this myself 54a516e7a52b60d77e98adab7d22d02346e5bea0 to fix this stuff sooner and tackle some other challenges from moving away from ffmpeg regex.

**Comment by WyattBlue** [comment]
I don't like making FFprobe a separate class, using FileInfo as a common interface makes much more since. Also your PR is not working on Linux. Probably because you're running "FFprobe" instead of "ffprobe" in subprocess.

I decided to implement this myself 54a516e7a52b60d77e98adab7d22d02346e5bea0 to fix this stuff sooner and tackle some other challenges from moving away from ffmpeg regex.
